model/driver/src/icon4py/model/driver/config/configuration_manager.py

```
@@ -0,0 +1,147 @@
+        run_config.update(user_config.run)  # type: ignore [union-attr]
+        self._handlers[RUN_NODE] = run_config
+        self._initialize_components(user_config)
+
+    # TODO (halungge): have this return a std dict to hide away omegaconf
+    def get(
+        self,
+        is_default: bool = False,
+    ) -> oc.DictConfig:
+        config_type = (
+            common_config.ConfigType.DEFAULT if is_default else common_config.ConfigType.CUSTOM
+        )
+        merged = oc.OmegaConf.create({})
+        oc.OmegaConf.set_readonly(merged, False)
+        for name, h in self._handlers.items():
+            merged[name] = h._get(
+                type_=config_type, format_=common_config.Format.DICT, read_only=False
+            )
+
+        oc.OmegaConf.resolve(merged)
+        oc.OmegaConf.set_readonly(merged, True)
+        return merged
+
+    def _initialize_components(self, config: oc.DictConfig) -> None:
+        model_components = config.model.components.items()
+        for name, package in model_components:
+            module_config = config.get(name)
+            try:
+                config_module = package + ".config"
+                module = importlib.import_module(config_module)
+                # merge the default of the package with the section configured in the file
+                reader = load_reader(module, module_config)
+            except ModuleNotFoundError:
+                log.warning(
+                    f"no config module in package {package}, instantiating general dict-config instead"
+                )
+                reader = common_config.init_config()
+                if module_config is None:
+                    log.warning(
+                        f"No default configuration and no user-configuration for component {name} defined in file {self._config_file}"
+                    )
+                else:
+                    reader.update(module_config)
```

It may be true in the if statement above that users have developed a component without any config parameters. 
However, I am wondering that should we just throw an error here in this case where users put a config for a particular component whereas there is in `config.py`.